#!/bin/bash


RGID="H3WNJBBXX.2"
RGLB="18"
RGPL="Illumina"
RGSM="RUD4_1_1"
RGPU="H3WNJBBXX.NTAGGC.2"
SampleID="RUD4_1_1"

Read1=""
Read2=""

Ref=""

## ngm mapping
Cibiv-NextGenMap-ce00bcf/bin/ngm-0.4.12/ngm -r $Ref -1 $Read1 -2 $Read2 -o "ngm."$SampleID".sam"

## sam to bam conversion
module load samtools
samtools view -bS -t $Ref -o "ngm."$SampleID".bam" "ngm."$SampleID".sam"

## picard-tools AddOrReplaceReadGroups
module load picard
java -jar picard-tools-1.136/picard.jar AddOrReplaceReadGroups I="ngm."$SampleID".bam" O="ngm."$SampleID".rg.bam" RGID=$RGID RGLB=$RGLB RGPL=$RGPL RGSM=$RGSM RGPU=$RGPU

# Sort Sam
java -jar picard-tools-1.136/picard.jar SortSam I="ngm."$SampleID".rg.bam" O="ngm."$SampleID".sorted.bam" SORT_ORDER=coordinate VALIDATION_STRINGENCY=LENIENT

#Build bam index
java -jar picard-tools-1.136/picard.jar BuildBamIndex I="ngm."$SampleID".sorted.bam" VALIDATION_STRINGENCY=LENIENT


# GATK, RealingerCreator
java -jar GenomeAnalysisTK.jar -T RealignerTargetCreator -I "ngm."$SampleID".sorted.bam" -R $Ref -o "ngm."$SampleID".sorted.intervals"



# GATK IndelRealinger 
java -jar GenomeAnalysisTK.jar -T IndelRealigner -I "ngm."$SampleID".sorted.bam" -R $Ref -targetIntervals "ngm."$SampleID".sorted.intervals" -o "ngm."$SampleID".sorted.real.bam"


# Mark Duplicates
java -jar picard-tools-1.136/picard.jar MarkDuplicates I="ngm."$SampleID".sorted.real.bam" REMOVE_DUPLICATES=true VALIDATION_STRINGENCY=LENIENT AS=true METRICS_FILE="ngm."$SampleID".sorted.real.dups" OUTPUT="ngm."$SampleID".sorted.real.dups.bam"

## Build bam index on mark duplicated bam file
java -jar picard-tools-1.136/picard.jar BuildBamIndex I="ngm."$SampleID".sorted.real.dups.bam" VALIDATION_STRINGENCY=LENIENT

## GATK HaplotypeCaller : make vcf for each sample
java -jar GenomeAnalysisTK.jar -l INFO -T HaplotypeCaller -R $Ref --emitRefConfidence GVCF --variant_index_type LINEAR --variant_index_parameter 128000 --genotyping_mode DISCOVERY -stand_emit_conf 10 -stand_call_conf 30 -nct 6 -o $SampleID".snp.GATK.raw.g.vcf" -I "ngm."$SampleID".sorted.real.dups.bam" 

# genotypeGVCF Gvcf.list contains a list of 107 vcf generated by HaplotypeCaller 
java -jar GenomeAnalysisTK.jar -R $Ref -T GenotypeGVCFs -o all.v3.6.raw_107line.vcf  --variant Gvcf.list -nt 6

#### Annotate with SnpEff
# Modify chromosome name
sed 's/^6/Pt/g' all.v3.6.raw_107line.vcf | sed 's/^7/Mt/g' > all.v3.6.raw_107line.PtMt.vcf &
# run SnpEff without up and downstream annotation
java -jar snpEff.jar -ud 0 -v TAIR10.29 all.v3.6.raw_107line.PtMt.vcf > all.v3.6.raw_107line.PtMt.ann.ud0.vcf
# change chromosome name back to number
sed 's/^Pt/6/g' all.v3.6.raw_107line.PtMt.ann.ud0.vcf |sed 's/^Mt/7/g' > all.v3.6.raw_107line.ann.ud0.vcf &


### select variants
# snp
java -jar GenomeAnalysisTK.jar -T SelectVariants -selectType SNP -R $Ref --variant all.v3.6.raw_107line.ann.ud0.vcf -o all.v3.6.raw_107line.ann.ud0.snp.vcf
# indel
java -jar GenomeAnalysisTK.jar -T SelectVariants -selectType INDEL -R $Ref --variant all.v3.6.raw_107line.ann.ud0.vcf -o all.v3.6.raw_107line.ann.ud0.indel.vcf


### Hard filter
# snp
java -jar GenomeAnalysisTK.jar -T VariantFiltration -R $Ref --variant all.v3.6.raw_107line.ann.ud0.snp.vcf --filterExpression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filterName "default_filter" -o all.v3.6.raw_107line.ann.ud0.snp.default_filter.vcf 
# indel
java -jar GenomeAnalysisTK.jar -T VariantFiltration -R $Ref --variant all.v3.6.raw_107line.ann.ud0.indel.vcf --filterExpression "QD < 2.0 || FS > 200.0  || ReadPosRankSum < -20.0" --filterName "default_filter" -o all.v3.6.raw_107line.ann.ud0.indel.default_filter.vcf 




# Identify mutations. The 9th column is AC and the 13th column is AN. Select variants that has AC=2 and AN>=107. Print out variants that are homozygous alternative genotype (1/1), and filter out variants with coverage less than 3 
grep -v "#" all.v3.6.raw_107line.ann.ud0.snp.vcf |awk -F "[\t;\=]" '$9==2 && $13 >= 107'|grep -v ^7|grep -v ^6 | awk '{for (i=10;i<=NF;i++) {if ($i ~ /1\/1/) {print i"\t"$i"\t"$0}}}' |awk '$2 !~ /0,1:/ && $2 !~ /0,2:/ && $2 !~ /0,0:/' |wc -l
# 1694

grep -v "#" all.v3.6.raw_107line.ann.ud0.indel.vcf |awk -F "[\t;\=]" '$9==2 && $13 >= 107'|grep -v ^7|grep -v ^6 | awk '{for (i=10;i<=NF;i++) {if ($i ~ /1\/1/) {print i"\t"$i"\t"$0}}}' |awk '$2 !~ /0,1:/ && $2 !~ /0,2:/ && $2 !~ /0,0:/' |grep -v AAAAAAAAAA |grep -v TTTTTTTTTT|wc -l
# 322

# long indel identified by Delly
#!/bin/bash -l

#SBATCH
#SBATCH --job-name=Delly_1R.S1
#SBATCH -o Delly_1R.S1.o%j
#SBATCH --time=24:0:0
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=24
#SBATCH --partition=shared

line="1R.S1"

#SV calling by sample
#~/delly_v0.7.6_parallel_linux_x86_64bit call -t DEL -g ../Ref/TAIR9_masked.fa -o $line"_DEL.bcf" ../GATK_done/ngm.$line".sorted.real.dups.bam" 
#~/delly_v0.7.6_parallel_linux_x86_64bit call -t DUP -g ../Ref/TAIR9_masked.fa -o $line"_DUP.bcf" ../GATK_done/ngm.$line".sorted.real.dups.bam" 
#~/delly_v0.7.6_parallel_linux_x86_64bit call -t INV -g ../Ref/TAIR9_masked.fa -o $line"_INV.bcf" ../GATK_done/ngm.$line".sorted.real.dups.bam" 
#~/delly_v0.7.6_parallel_linux_x86_64bit call -t TRA -g ../Ref/TAIR9_masked.fa -o $line"_TRA.bcf" ../GATK_done/ngm.$line".sorted.real.dups.bam" 
#~/delly_v0.7.6_parallel_linux_x86_64bit call -t INS -g ../Ref/TAIR9_masked.fa -o $line"_INS.bcf" ../GATK_done/ngm.$line".sorted.real.dups.bam" 

#Merge SV sites into a unified site list
#~/delly_v0.7.6_parallel_linux_x86_64bit merge -t DEL -m 2 -o del.bcf 100.S14_DEL.bcf 101.S15_DEL.bcf 102.S16_DEL.bcf 103.S17_DEL.bcf 105.S1_DEL.bcf 106.S2_DEL.bcf 108_DEL.bcf 109R.S4_DEL.bcf 110R.S5_DEL.bcf 111.S6_DEL.bcf 112R2.S7_DEL.bcf 113.S8_DEL.bcf 114.S9_DEL.bcf 115.S10_DEL.bcf 116.S11_DEL.bcf 117.S12_DEL.bcf 118.S13_DEL.bcf 119.bcf 11.S2_DEL.bcf 12B.S9_DEL.bcf 13.S10_DEL.bcf 14.S3_DEL.bcf 15R.S12_DEL.bcf 16.S13_DEL.bcf 17.S14_DEL.bcf 18.S15_DEL.bcf 19.S4_DEL.bcf 1R.S1_DEL.bcf 20R2.S5_DEL.bcf 22.S16_DEL.bcf 23.S17_DEL.bcf 24.S18_DEL.bcf 25.S6_DEL.bcf 26.S1_DEL.bcf 27RB.S2_DEL.bcf 28.S3_DEL.bcf 29R.S4_DEL.bcf 2R.S2_DEL.bcf 31.S5_DEL.bcf 32.S6_DEL.bcf 33.S19_DEL.bcf 34.S20_DEL.bcf 35.S7_DEL.bcf 36.S8_DEL.bcf 37R.S9_DEL.bcf 38.S10_DEL.bcf 39.S10_DEL.bcf 3.S3_DEL.bcf 40.S13_DEL.bcf 41.S13_DEL.bcf 42.S14_DEL.bcf 43.S15_DEL.bcf 44B.S16_DEL.bcf 45.S17_DEL.bcf 46.S4_DEL.bcf 47R2.S5_DEL.bcf 48.S18_DEL.bcf 49_DEL.bcf 4.S4_DEL.bcf 50R2.S20_DEL.bcf 51.S1_DEL.bcf 52.S2_DEL.bcf 53R.S3_DEL.bcf 54.S4_DEL.bcf 55.S5_DEL.bcf 56.S6_DEL.bcf 57R2.S7_DEL.bcf 58.S8_DEL.bcf 59R2.S9_DEL.bcf 5.S5_DEL.bcf 60.S10_DEL.bcf 61.S11_DEL.bcf 62R.S6_DEL.bcf 63.S7_DEL.bcf 64.S8_DEL.bcf 65.S9_DEL.bcf 66R.S12_DEL.bcf 67.S13_DEL.bcf 68R.S14_DEL.bcf 69.S15_DEL.bcf 6.S6_DEL.bcf 71BR.S16_DEL.bcf 72.S17_DEL.bcf 73.S18_DEL.bcf 74.S19_DEL.bcf 75.S20_DEL.bcf 76.S7_DEL.bcf 77.S10_DEL.bcf 78R.S8_DEL.bcf 79.S11_DEL.bcf 7.S7_DEL.bcf 80.S12_DEL.bcf 81.S13_DEL.bcf 82.S14_DEL.bcf 83.S15_DEL.bcf 84R.S16_DEL.bcf 85.S9_DEL.bcf 86.S10_DEL.bcf 88R2.S11_DEL.bcf 89R.S12_DEL.bcf 8R.S8_DEL.bcf 91.S13_DEL.bcf 92.S14_DEL.bcf 94.S15_DEL.bcf 96.S16_DEL.bcf 98.S17_DEL.bcf 99.S18_DEL.bcf 9.S1_DEL.bcf 

#~/delly_v0.7.6_parallel_linux_x86_64bit merge -t DUP -m 2 -o dup.bcf 100.S14_DUP.bcf 101.S15_DUP.bcf 102.S16_DUP.bcf 103.S17_DUP.bcf 105.S1_DUP.bcf 106.S2_DUP.bcf 108.S3_DUP.bcf 109R.S4_DUP.bcf 110R.S5_DUP.bcf 111.S6_DUP.bcf 112R2.S7_DUP.bcf 113.S8_DUP.bcf 114.S9_DUP.bcf 115.S10_DUP.bcf 116.S11_DUP.bcf 117.S12_DUP.bcf 118.S13_DUP.bcf 119.S14_DUP.bcf 11.S2_DUP.bcf 12B.S9_DUP.bcf 13.S10_DUP.bcf 14.S3_DUP.bcf 15R.S12_DUP.bcf 16.S13_DUP.bcf 17.S14_DUP.bcf 18.S15_DUP.bcf 19.S4_DUP.bcf 1R.S1_DUP.bcf 20R2.S5_DUP.bcf 22.S16_DUP.bcf 23.S17_DUP.bcf 24.S18_DUP.bcf 25.S6_DUP.bcf 26.S1_DUP.bcf 27RB.S2_DUP.bcf 28.S3_DUP.bcf 29R.S4_DUP.bcf 2R.S2_DUP.bcf 31.S5_DUP.bcf 32.S6_DUP.bcf 33.S19_DUP.bcf 34.S20_DUP.bcf 35.S7_DUP.bcf 36.S8_DUP.bcf 37R.S9_DUP.bcf 38.S10_DUP.bcf 39.S10_DUP.bcf 3.S3_DUP.bcf 40.S13_DUP.bcf 41.S13_DUP.bcf 42.S14_DUP.bcf 43.S15_DUP.bcf 44B.S16_DUP.bcf 45.S17_DUP.bcf 46.S4_DUP.bcf 47R2.S5_DUP.bcf 48.S18_DUP.bcf 49.S19_DUP.bcf 4.S4_DUP.bcf 50R2.S20_DUP.bcf 51.S1_DUP.bcf 52.S2_DUP.bcf 53R.S3_DUP.bcf 54.S4_DUP.bcf 55.S5_DUP.bcf 56.S6_DUP.bcf 57R2.S7_DUP.bcf 58.S8_DUP.bcf 59R2.S9_DUP.bcf 5.S5_DUP.bcf 60.S10_DUP.bcf 61.S11_DUP.bcf 62R.S6_DUP.bcf 63.S7_DUP.bcf 64.S8_DUP.bcf 65.S9_DUP.bcf 66R.S12_DUP.bcf 67.S13_DUP.bcf 68R.S14_DUP.bcf 69.S15_DUP.bcf 6.S6_DUP.bcf 71BR.S16_DUP.bcf 72.S17_DUP.bcf 73.S18_DUP.bcf 74.S19_DUP.bcf 75.S20_DUP.bcf 76.S7_DUP.bcf 77.S10_DUP.bcf 78R.S8_DUP.bcf 79.S11_DUP.bcf 7.S7_DUP.bcf 80.S12_DUP.bcf 81.S13_DUP.bcf 82.S14_DUP.bcf 83.S15_DUP.bcf 84R.S16_DUP.bcf 85.S9_DUP.bcf 86.S10_DUP.bcf 88R2.S11_DUP.bcf 89R.S12_DUP.bcf 8R.S8_DUP.bcf 91.S13_DUP.bcf 92.S14_DUP.bcf 94.S15_DUP.bcf 96.S16_DUP.bcf 98.S17_DUP.bcf 99.S18_DUP.bcf 9.S1_DUP.bcf
#~/delly_v0.7.6_parallel_linux_x86_64bit merge -t INS -m 2 -o ins.bcf 100.S14_INS.bcf 101.S15_INS.bcf 102.S16_INS.bcf 103.S17_INS.bcf 105.S1_INS.bcf 106.S2_INS.bcf 108.S3_INS.bcf 109R.S4_INS.bcf 110R.S5_INS.bcf 111.S6_INS.bcf 112R2.S7_INS.bcf 113.S8_INS.bcf 114.S9_INS.bcf 115.S10_INS.bcf 116.S11_INS.bcf 117.S12_INS.bcf 118.S13_INS.bcf 119.S14_INS.bcf 11.S2_INS.bcf 12B.S9_INS.bcf 13.S10_INS.bcf 14.S3_INS.bcf 15R.S12_INS.bcf 16.S13_INS.bcf 17.S14_INS.bcf 18.S15_INS.bcf 19.S4_INS.bcf 1R.S1_INS.bcf 20R2.S5_INS.bcf 22.S16_INS.bcf 23.S17_INS.bcf 24.S18_INS.bcf 25.S6_INS.bcf 26.S1_INS.bcf 27RB.S2_INS.bcf 28.S3_INS.bcf 29R.S4_INS.bcf 2R.S2_INS.bcf 31.S5_INS.bcf 32.S6_INS.bcf 33.S19_INS.bcf 34.S20_INS.bcf 35.S7_INS.bcf 36.S8_INS.bcf 37R.S9_INS.bcf 38.S10_INS.bcf 39.S10_INS.bcf 3.S3_INS.bcf 40.S13_INS.bcf 41.S13_INS.bcf 42.S14_INS.bcf 43.S15_INS.bcf 44B.S16_INS.bcf 45.S17_INS.bcf 46.S4_INS.bcf 47R2.S5_INS.bcf 48.S18_INS.bcf 49.S19_INS.bcf 4.S4_INS.bcf 50R2.S20_INS.bcf 51.S1_INS.bcf 52.S2_INS.bcf 53R.S3_INS.bcf 54.S4_INS.bcf 55.S5_INS.bcf 56.S6_INS.bcf 57R2.S7_INS.bcf 58.S8_INS.bcf 59R2.S9_INS.bcf 5.S5_INS.bcf 60.S10_INS.bcf 61.S11_INS.bcf 62R.S6_INS.bcf 63.S7_INS.bcf 64.S8_INS.bcf 65.S9_INS.bcf 66R.S12_INS.bcf 67.S13_INS.bcf 68R.S14_INS.bcf 69.S15_INS.bcf 6.S6_INS.bcf 71BR.S16_INS.bcf 72.S17_INS.bcf 73.S18_INS.bcf 74.S19_INS.bcf 75.S20_INS.bcf 76.S7_INS.bcf 77.S10_INS.bcf 78R.S8_INS.bcf 79.S11_INS.bcf 7.S7_INS.bcf 80.S12_INS.bcf 81.S13_INS.bcf 82.S14_INS.bcf 83.S15_INS.bcf 84R.S16_INS.bcf 85.S9_INS.bcf 86.S10_INS.bcf 88R2.S11_INS.bcf 89R.S12_INS.bcf 8R.S8_INS.bcf 91.S13_INS.bcf 92.S14_INS.bcf 94.S15_INS.bcf 96.S16_INS.bcf 98.S17_INS.bcf 99.S18_INS.bcf 9.S1_INS.bcf 
#~/delly_v0.7.6_parallel_linux_x86_64bit merge -t INV -m 2 -o inv.bcf 100.S14_INV.bcf 101.S15_INV.bcf 102.S16_INV.bcf 103.S17_INV.bcf 105.S1_INV.bcf 106.S2_INV.bcf 108.S3_INV.bcf 109R.S4_INV.bcf 110R.S5_INV.bcf 111.S6_INV.bcf 112R2.S7_INV.bcf 113.S8_INV.bcf 114.S9_INV.bcf 115.S10_INV.bcf 116.S11_INV.bcf 117.S12_INV.bcf 118.S13_INV.bcf 119.S14_INV.bcf 11.S2_INV.bcf 12B.S9_INV.bcf 13.S10_INV.bcf 14.S3_INV.bcf 15R.S12_INV.bcf 16.S13_INV.bcf 17.S14_INV.bcf 18.S15_INV.bcf 19.S4_INV.bcf 1R.S1_INV.bcf 20R2.S5_INV.bcf 22.S16_INV.bcf 23.S17_INV.bcf 24.S18_INV.bcf 25.S6_INV.bcf 26.S1_INV.bcf 27RB.S2_INV.bcf 28.S3_INV.bcf 29R.S4_INV.bcf 2R.S2_INV.bcf 31.S5_INV.bcf 32.S6_INV.bcf 33.S19_INV.bcf 34.S20_INV.bcf 35.S7_INV.bcf 36.S8_INV.bcf 37R.S9_INV.bcf 38.S10_INV.bcf 39.S10_INV.bcf 3.S3_INV.bcf 40.S13_INV.bcf 41.S13_INV.bcf 42.S14_INV.bcf 43.S15_INV.bcf 44B.S16_INV.bcf 45.S17_INV.bcf 46.S4_INV.bcf 47R2.S5_INV.bcf 48.S18_INV.bcf 49.S19_INV.bcf 4.S4_INV.bcf 50R2.S20_INV.bcf 51.S1_INV.bcf 52.S2_INV.bcf 53R.S3_INV.bcf 54.S4_INV.bcf 55.S5_INV.bcf 56.S6_INV.bcf 57R2.S7_INV.bcf 58.S8_INV.bcf 59R2.S9_INV.bcf 5.S5_INV.bcf 60.S10_INV.bcf 61.S11_INV.bcf 62R.S6_INV.bcf 63.S7_INV.bcf 64.S8_INV.bcf 65.S9_INV.bcf 66R.S12_INV.bcf 67.S13_INV.bcf 68R.S14_INV.bcf 69.S15_INV.bcf 6.S6_INV.bcf 71BR.S16_INV.bcf 72.S17_INV.bcf 73.S18_INV.bcf 74.S19_INV.bcf 75.S20_INV.bcf 76.S7_INV.bcf 77.S10_INV.bcf 78R.S8_INV.bcf 79.S11_INV.bcf 7.S7_INV.bcf 80.S12_INV.bcf 81.S13_INV.bcf 82.S14_INV.bcf 83.S15_INV.bcf 84R.S16_INV.bcf 85.S9_INV.bcf 86.S10_INV.bcf 88R2.S11_INV.bcf 89R.S12_INV.bcf 8R.S8_INV.bcf 91.S13_INV.bcf 92.S14_INV.bcf 94.S15_INV.bcf 96.S16_INV.bcf 98.S17_INV.bcf 99.S18_INV.bcf 9.S1_INV.bcf 
#~/delly_v0.7.6_parallel_linux_x86_64bit merge -t TRA -m 2 -o tra.bcf 100.S14_TRA.bcf 101.S15_TRA.bcf 102.S16_TRA.bcf 103.S17_TRA.bcf 105.S1_TRA.bcf 106.S2_TRA.bcf 108.S3_TRA.bcf 109R.S4_TRA.bcf 110R.S5_TRA.bcf 111.S6_TRA.bcf 112R2.S7_TRA.bcf 113.S8_TRA.bcf 114.S9_TRA.bcf 115.S10_TRA.bcf 116.S11_TRA.bcf 117.S12_TRA.bcf 118.S13_TRA.bcf 119.S14_TRA.bcf 11.S2_TRA.bcf 12B.S9_TRA.bcf 13.S10_TRA.bcf 14.S3_TRA.bcf 15R.S12_TRA.bcf 16.S13_TRA.bcf 17.S14_TRA.bcf 18.S15_TRA.bcf 19.S4_TRA.bcf 1R.S1_TRA.bcf 20R2.S5_TRA.bcf 22.S16_TRA.bcf 23.S17_TRA.bcf 24.S18_TRA.bcf 25.S6_TRA.bcf 26.S1_TRA.bcf 27RB.S2_TRA.bcf 28.S3_TRA.bcf 29R.S4_TRA.bcf 2R.S2_TRA.bcf 31.S5_TRA.bcf 32.S6_TRA.bcf 33.S19_TRA.bcf 34.S20_TRA.bcf 35.S7_TRA.bcf 36.S8_TRA.bcf 37R.S9_TRA.bcf 38.S10_TRA.bcf 39.S10_TRA.bcf 3.S3_TRA.bcf 40.S13_TRA.bcf 41.S13_TRA.bcf 42.S14_TRA.bcf 43.S15_TRA.bcf 44B.S16_TRA.bcf 45.S17_TRA.bcf 46.S4_TRA.bcf 47R2.S5_TRA.bcf 48.S18_TRA.bcf 49.S19_TRA.bcf 4.S4_TRA.bcf 50R2.S20_TRA.bcf 51.S1_TRA.bcf 52.S2_TRA.bcf 53R.S3_TRA.bcf 54.S4_TRA.bcf 55.S5_TRA.bcf 56.S6_TRA.bcf 57R2.S7_TRA.bcf 58.S8_TRA.bcf 59R2.S9_TRA.bcf 5.S5_TRA.bcf 60.S10_TRA.bcf 61.S11_TRA.bcf 62R.S6_TRA.bcf 63.S7_TRA.bcf 64.S8_TRA.bcf 65.S9_TRA.bcf 66R.S12_TRA.bcf 67.S13_TRA.bcf 68R.S14_TRA.bcf 69.S15_TRA.bcf 6.S6_TRA.bcf 71BR.S16_TRA.bcf 72.S17_TRA.bcf 73.S18_TRA.bcf 74.S19_TRA.bcf 75.S20_TRA.bcf 76.S7_TRA.bcf 77.S10_TRA.bcf 78R.S8_TRA.bcf 79.S11_TRA.bcf 7.S7_TRA.bcf 80.S12_TRA.bcf 81.S13_TRA.bcf 82.S14_TRA.bcf 83.S15_TRA.bcf 84R.S16_TRA.bcf 85.S9_TRA.bcf 86.S10_TRA.bcf 88R2.S11_TRA.bcf 89R.S12_TRA.bcf 8R.S8_TRA.bcf 91.S13_TRA.bcf 92.S14_TRA.bcf 94.S15_TRA.bcf 96.S16_TRA.bcf 98.S17_TRA.bcf 99.S18_TRA.bcf 9.S1_TRA.bcf 

#Re-genotype merged SV site list across all samples.
#~/delly_v0.7.6_parallel_linux_x86_64bit call -t DEL -g ../Ref/TAIR9_masked.fa -v del.bcf -o $line"_DEL.geno.bcf" ../GATK_done/ngm.$line".sorted.real.dups.bam" 

~/delly_v0.7.6_parallel_linux_x86_64bit call -t DUP -g ../Ref/TAIR9_masked.fa -v dup.bcf -o $line"_DUP.geno.bcf" ../GATK_done/ngm.$line".sorted.real.dups.bam" 
~/delly_v0.7.6_parallel_linux_x86_64bit call -t INS -g ../Ref/TAIR9_masked.fa -v ins.bcf -o $line"_INS.geno.bcf" ../GATK_done/ngm.$line".sorted.real.dups.bam" 
~/delly_v0.7.6_parallel_linux_x86_64bit call -t INV -g ../Ref/TAIR9_masked.fa -v inv.bcf -o $line"_INV.geno.bcf" ../GATK_done/ngm.$line".sorted.real.dups.bam" 
~/delly_v0.7.6_parallel_linux_x86_64bit call -t TRA -g ../Ref/TAIR9_masked.fa -v tra.bcf -o $line"_TRA.geno.bcf" ../GATK_done/ngm.$line".sorted.real.dups.bam" 


#Merge all re-genotyped samples to get a single VCF/BCF using bcftools merge
#module load bcftools
#bcftools merge -O b -o merged_49_119.bcf 49_DEL.geno.bcf 119_DEL.geno.bcf

#############


# Dec 9,2016
# Unique mutation among 108 lines at chromosome 1-5 (AC=2, AN>=108, without HET) = 1865

grep -v "#" all.v3.6.raw_108line.snp.vcf|awk -F "[\t;\=]" '$9==2 && $13 >= 108'|grep -v "0/1" |grep -v ^7|grep -v ^6 > all.v3.6.raw_108line.snp_AC2AN108.vcf



# Unique mutation among 108 lines at chromosome 1-5 (AC=2, without HET) = 2464

grep -v "#" all.v3.6.raw_108line.snp.vcf|awk -F "[\t;\=]" '$9==2 '|grep -v "0/1" |grep -v ^7|grep -v ^6 > all.v3.6.raw_108line.snp_AC2.vcf

# Print the MA line with mutation “1/1” and exclude low coverage calls, DP=0,1,2  
# 1734

awk '{for (i=1;i<=NF;i++) {if ($i ~ /1\/1/) {print i"\t"$i"\t"$0}}}' all.v3.6.raw_108line.snp_AC2.vcf |awk '$2 !~ /0,1:/ && $2 !~ /0,2:/ && $2 !~ /0,0:/'> all.v3.6.raw_108line.snp_AC2_DP3.vcf





#genotypeGVCF -allSite, call variant and non-variant sites for each line
java -jar GenomeAnalysisTK.jar -R $Ref -T GenotypeGVCFs -o $SampleID".v3.6.gVCF.allSite.vcf"109R.GATK.genotype.allSite.vcf.gz -allSites --variant $SampleID".snp.GATK.raw.g.vcf" -nt 6


# 5 parental lines (P14, P7, P723, P87,P1)
java -jar $HOME/GenomeAnalysisTK-3.6-0-g89b7209/GenomeAnalysisTK.jar -R /home1/05573/tg848723/Ref/TAIR9_masked.fa -T GenotypeGVCFs -o ShawF.all5.GATK.genotype.vcf.gz  --variant ShawF.p1.snp.GATK.raw.g.vcf --variant ShawF.P723.snp.GATK.raw.g.vcf --variant ShawF.P14.snp.GATK.raw.g.vcf --variant ShawF.P7.snp.GATK.raw.g.vcf  --variant ShawF.P87.snp.GATK.raw.g.vcf -nt 12













wc -l /Volumes/Seagate_8TB_backup/GATK_all.v3.6_107lines/all.v3.6.raw_107line.ann.ud0.indel.default_filter.AC2AN107_DP3.vcf 
     393 
wc -l /Volumes/Seagate_8TB_backup/GATK_all.v3.6_107lines/all.v3.6.raw_107line.ann.ud0.snp.default_filter.AC2AN107_DP3.vcf 
    1694 